### PR TITLE
fix(launcher): running getMultiCapabilities should reject on errors

### LINKED ISSUE #3875: exit code 0 on failure (specifically getMultiCapabilities failure)
In my team's setup, we managed more than once to miss out on protractor not actually running, although we do follow proper exit codes.

Yet protractor does *not*.

Our latest mishap was a failure in our `getMultiCapabilities` function. Protractor does not reject/catch, meaning errors get silenced and protractor fails to run any test, but exits with 0, and our CI trusts that.

### PR DESCRIPTION
**Pull request by cnishina**
closes #3875

### CODE REVIEW

File: lib/launcher.ts [hunk 1]
```diff
@@ -123,10 +123,18 @@ let initFn = function(configFile: string, additionalConfig: Config) {
                       'and multiCapabilities');
                 }
                 // If getMultiCapabilities is defined and a function, use this.
-                q.when(config.getMultiCapabilities(), (multiCapabilities) => {
-                   config.multiCapabilities = multiCapabilities;
-                   config.capabilities = null;
-                 }).then(() => resolve());
+                q(config.getMultiCapabilities())
+                    .then((multiCapabilities) => {
+                      console.log('here');
```
Comment (mgiambalvo) [inline lib/launcher.ts]: Leftover console.log

File: lib/launcher.ts [hunk 2]
```diff
@@ -112,7 +112,7 @@ let initFn = function(configFile: string, additionalConfig: Config) {
       .then(() => {
 
         return q
-            .Promise<any>((resolve: Function) => {
+            .Promise<any>((resolve: Function, reject: Function) => {
```
Comment (sjelin) [inline lib/launcher.ts]: Should be `new q.Promise(...`.  Not your fault but please fix anyway

File: scripts/errorTest.js [hunk 3]
```diff
@@ -44,3 +44,11 @@ messages = [
   'Cannot run in debug mode with multiCapabilities, count > 1, or sharding',
   'Process exited with error code ' + exitCodes.ConfigError.CODE];
 checkLogs(output, messages);
+
+runProtractor = spawn('node',
+    ['bin/protractor', 'spec/errorTest/getMultiCapabilitiesConf.js']);
+output = runProtractor.stderr.toString();
+console.log(output);
```
Comment (sjelin) [inline scripts/errorTest.js]: leftover from debugging?

File: spec/errorTest/getMultiCapabilitiesConf.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+  // Spec patterns are relative to this directory.
+  specs: [
+    '../../basic/mock*'
+  ],
+
+  framework: 'debugprint',
+  getMultiCapabilities: function() {
+    return new Promise((resolve) => {
+      this.foo.bar = 1; // will fail because this.foo is undefined
```
Comment (sjelin) [inline spec/errorTest/getMultiCapabilitiesConf.js]: I'd rather you just threw something

File: lib/launcher.ts (hunk 2 above)
Comment (cnishina) [inline lib/launcher.ts]: This is not possible.  Setting this to void returns a transpile error. `lib/launcher.ts(114,16): error TS2350: Only a void function can be called with the 'new' keyword.`

File: lib/launcher.ts (hunk 1 above)
Comment (cnishina) [inline lib/launcher.ts]: Thanks. Done.

File: scripts/errorTest.js (hunk 3 above)
Comment (cnishina) [inline scripts/errorTest.js]: Thanks. Done.

File: spec/errorTest/getMultiCapabilitiesConf.js (hunk 4 above)
Comment (cnishina) [inline spec/errorTest/getMultiCapabilitiesConf.js]: Done. Throwing error.

**Pull request merged**